### PR TITLE
removing old file types

### PR DESCRIPTION
**Pull request by truhacevkir**
A lot of changes in test setups. I see we have a ton of duplicated test methods over whole project when constructing files, I will do something with this in next pull requests. 

Removing old file types.

- In business logic substituting `AdministrativeAgreement` -> `InternalFile`.
- In tests substituting `PublishedFile` -> `OpenFile` and `UnpublishedFile` -> `PendingOpenFile`. 

Next steps will be:
- Remove `administrativeAgreement` boolean parameter. 
- Remove all file conversion methods as `toInternalFile()`, `toOpenFile()` from File.class to each subclass of File.class.

### CODE REVIEW

File: publication-model-testing/src/main/java/no/unit/nva/model/testing/associatedartifacts/UnpublishedFileGenerator.java
Comment (truhacevkir) [inline publication-model-testing/src/main/java/no/unit/nva/model/testing/associatedartifacts/UnpublishedFileGenerator.java:1]: This generator is not needed as we have AssociatedArtifactsGenerator.

File: publication-commons/src/main/java/no/unit/nva/publication/rightsretention/RightsRetentionsValueFinder.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public RightsRetentionStrategy getRightsRetentionStrategy(File file, Publication
 
     private boolean rrsIsIrrelevant(File file, Publication publication) {
         return !PublisherVersion.ACCEPTED_VERSION.equals(file.getPublisherVersion())
-               || file instanceof AdministrativeAgreement
+               || file instanceof InternalFile
                || !isAcademicArticle(publication);
     }
```
Comment (truhacevkir) [inline publication-commons/src/main/java/no/unit/nva/publication/rightsretention/RightsRetentionsValueFinder.java:59]: One of few changes in business logic, someone has to verify that it is correct change.

File: brage-import/src/test/java/no/sikt/nva/brage/migration/merger/CristinImportPublicationMergerTest.java [hunk 2]
```diff
@@ -587,11 +580,11 @@ var record = new Record();
     void shouldAddDublinCoreFromIncomingPublicationToExistingOneWhenItIsPresentInNewPublicationOnly()
         throws InvalidIssnException, InvalidIsbnException, InvalidUnconfirmedSeriesException {
         var existingPublication = randomPublication(AcademicMonograph.class);
-        var existingAssociatedArtifact = List.of(randomPublishedFile());
+        var existingAssociatedArtifact = List.<AssociatedArtifact>of(AssociatedArtifactsGenerator.randomOpenFile());
         existingPublication.setAssociatedArtifacts(new AssociatedArtifactList(existingAssociatedArtifact));
         var newPublication = randomPublication(AcademicMonograph.class);
         var newDublinCore = randomDublinCore();
-        var newAssociatedArtifact = List.of(newDublinCore, randomPublishedFile());
+        var newAssociatedArtifact = List.of(newDublinCore, AssociatedArtifactsGenerator.randomOpenFile());
```
Comment (brinxmat) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/merger/CristinImportPublicationMergerTest.java:587]: Please use fully qualified static imports e.g. simply `randomOpenFile()`

File: publication-commons/src/test/java/no/unit/nva/publication/rightsretention/RightsRetentionsApplierTest.java [hunk 3]
```diff
@@ -249,8 +249,8 @@ void shouldAllowPublishingCuratorToOverrideRetentionStrategy(RightsRetentionStra
         assertDoesNotThrow(applier::handle);
     }
 
-    private static UnpublishedFile createAcceptedFileWithRrs(RightsRetentionStrategy rrs) {
-        return createFileWithRrs(UUID.randomUUID(), ACCEPTED_VERSION, rrs);
+    private static PendingOpenFile createPendingOpenFileWithAcceptedVersionAndRrs(RightsRetentionStrategy rrs) {
+        return createPendingOpenFileWithRrs(UUID.randomUUID(), ACCEPTED_VERSION, rrs);
     }
 
     //            case OverriddenRightsRetentionStrategy strategy -> Set.of(OVERRIDABLE_RIGHTS_RETENTION_STRATEGY);
```
Comment (brinxmat) [inline publication-commons/src/test/java/no/unit/nva/publication/rightsretention/RightsRetentionsApplierTest.java:256]: Why did the commented code survive?

File: publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerRightsRetentionTest.java [hunk 4]
```diff
@@ -130,8 +130,8 @@ void shouldPreserveRrsOverridenByWhenChangingNonRrsFileMetadata() throws BadRequ
             OVERRIDABLE_RIGHTS_RETENTION_STRATEGY, rrsOverriddenBy
         );
         var fileId = UUID.randomUUID();
-        var file = createFileWithRrs(fileId, userSetRrs);
-        var updatedFile = createFileWithRrs(fileId, userSetRrs);
+        var file = createPendingOpenFileWithRrs(fileId, userSetRrs);
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerRightsRetentionTest.java:133]: PendingOpen?

**Review by brinxmat — COMMENTED** [review]
We still have the boolean administrative agreement? Do we need it?

**Comment by truhacevkir** [comment]
> //            case OverriddenRightsRetentionStrategy strategy -> Set.of(OVERRIDABLE_RIGHTS_RETENTION_STRATEGY);

Yes, we do. Will be removed in next pull requests.

File: publication-commons/src/test/java/no/unit/nva/publication/rightsretention/RightsRetentionsApplierTest.java (hunk 3 above)
Comment (truhacevkir) [reply publication-commons/src/test/java/no/unit/nva/publication/rightsretention/RightsRetentionsApplierTest.java:256]: Will remove it

File: publication-commons/src/main/java/no/unit/nva/publication/rightsretention/RightsRetentionsValueFinder.java (hunk 1 above)
Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/rightsretention/RightsRetentionsValueFinder.java:59]: HiddenFile should also probably be here?

**Review by brinxmat — APPROVED** [review]

File: publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ResourceOwnerPermissionStrategyTest.java [hunk 5]
```diff
@@ -132,41 +134,15 @@ void shoulDenyResourceOwnerOnDegree(PublicationOperation operation)
...
-                                                      .withLicense(PublicationGenerator.randomUri())
-                                                      .buildInternalFile()))))
+            arguments(named(OpenFile.TYPE, List.of(randomOpenFile())),
+            arguments(named(InternalFile.TYPE, List.of(randomInternalFile()))))
         );
     }
 
     public static Stream<Arguments> filesWithNotApprovedStatus() {
         return Stream.of(
-            arguments(named("UnpublishedFile", List.of(File.builder()
```
Comment (torbjokv) [inline publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ResourceOwnerPermissionStrategyTest.java:165]: no pending file here?

**Review by torbjokv — APPROVED** [review]
looks good! We should probably make sure we notify our api-consumers before pushing this to test

File: publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ResourceOwnerPermissionStrategyTest.java (hunk 5 above)
Comment (truhacevkir) [reply publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ResourceOwnerPermissionStrategyTest.java:165]: Adding pending files to this one.

**Pull request merged**